packages/legend-query/src/components/QueryBuilderValueSpecificationEditor.tsx

```diff
@@ -617,11 +1180,22 @@ export const DateInstanceValueEditor = observer(
  */
 export const QueryBuilderValueSpecificationEditor: React.FC<{
   valueSpecification: ValueSpecification;
+  queryBuilderState: QueryBuilderState;
   graph: PureModel;
   expectedType: Type;
   className?: string | undefined;
+  setValueSpecification: (val: ValueSpecification | undefined) => void;
+  matchType?: boolean | undefined;
```

Comment:
I don't think I get the point of having `matchType`. We should document it, and perhaps remove it altogether. Let's discuss about that. Anyway, we should make `expectedType` optional as well.

Reply:
As discussed, `matchedType` behavior can be controlled using `expectedType`, for example, when we're dealing with filter, we can say if the type of the field is `DateTime`, we can say expected type is `Date`, for example, thanks to stuff like `IS_ON_DAY`.